### ISSUE TITLE
Integrate Search Filters related components into lessons workflow

### ISSUE THREAD

**Issue by marcellamaki** [learningequality/kolibri#12788]
## Overview

The lesson management workflow is being aligned to use the similar side panel management UX that we introduced in quizzes in 0.17. As we work on extending some of the technical aspects of that work (managing the selection of resources and folders, tracking and updating local state, etc.), we will work in parallel to build out the components and pages that surround those parts of the workflow. 

This issue is to integrate [the updated `<SearchFiltersPanel>`](https://github.com/learningequality/kolibri/issues/12521) into lessons. The new temporary routes, added in https://github.com/learningequality/kolibri/pull/12672 should be used. 

### Prerequisites
Depends on #12521. In preparation for this issue, please add yourself as a reviewer of the [PR](https://github.com/learningequality/kolibri/pull/12759).  Catch up on the conversation, and also review the PR before you get started. 

### UX
![Image](https://github.com/user-attachments/assets/f00f9591-0644-4228-9f10-5d265486a0c7)
![Image](https://github.com/user-attachments/assets/69ed4f17-219d-4464-a8c9-2a56ad045c7b)

### Acceptance Criteria 
- [ ] Follow the implementation of displaying the search/filters panel within lessons, which is connected to the use of [the `useSearch()` composable](https://github.com/learningequality/kolibri/issues/12516).
- [ ] Note that in "quizzes", the activity selection is not displayed, whereas within lessons, the activity type is required
- [ ] Do not refactor `useSearch()`

Optional: 
- [ ] If #12786 is merged when this is in PR, please rebase and connect the opening of the search filters panel to the button

### Out of scope
- [ ] Fully integrating `useSearch()` in lessons which requires rendering a selectable list of search results. There are interdependencies on the resource selection work being decided in parallel. **Once the initial integration of the panel is completed, please check in with @marcellamaki to determine next steps about rendering cards for the search results, and what our plan will be for manual QA.**

## Resources
- [Figma](https://www.figma.com/design/lVJt5ukOrxS9qay0rXy7GC/0.18-Coach-updates?node-id=107-20606&node-type=canvas&t=BtL8UCkGoj5bhl9p-0)
